mbed-client/m2mbase.h

```diff
@@ -254,13 +280,13 @@ class M2MBase : public M2MReportObserver {
      * \brief Returns the interface description of the object.
      * \return The interface description of the object.
      */
-    virtual const String& interface_description() const;
+    virtual const char* interface_description() const;
 
     /**
      * \brief Returns the resource type of the object.
      * \return The resource type of the object.
      */
-    virtual const String& resource_type() const;
+    virtual const char* resource_type() const;
```

Comment:
Yes, this change will break the backward compatibility. In case we want to support old API we need to keep String members as well and that will of course increase the class size.